### PR TITLE
Feature: Set up navigational items to have an understanding of active links

### PR DESCRIPTION
**Pull request by edwinjoseph**
## Description
I want to be able to set a navigational item to be active when the page that I am on matches the url of the item. However, if the item is within a subtree, the path to the item also needs to be set to active. This way I am able to set styling that allows users to clearly understand what page they are currently viewing.

To do this, each navigation item has context of the path taken to get to itself and when we need to set the active route we take that absolute path and use it as the value.

## Acceptance Criteria 
- [x] When calling the property `setActiveRoute` from a navigation item, the active route is set with absolute path's value

### CODE REVIEW

File: example/src/App.js [hunk 1]
```diff
@@ -8,33 +9,60 @@ export const Nav = withNav(
...
-    </NavItem>
-  </Nav>
+    </header>
+    <main>
+      <Route exact path="/" render={page("Home")} />
+      <Route exact path="/about" render={page("About")} />
+      <Route exact path="/code" render={page("Code")} />
+      <Route exact path="/contact" render={page("Contact")} />
+    </main>
+  </Router>
```
[the remark this answers is not in the document]
Comment (edwinjoseph) [reply example/src/App.js]: Might be good to use Storybook?

File: .gitignore [hunk 2]
```diff
@@ -10,6 +10,7 @@ dist
 .rpt2_cache
 
 # misc
+.idea
```
[the remark this answers is not in the document]
Comment (edwinjoseph) [reply .gitignore]: 👍

File: src/utils/build-render-props.js [hunk 3]
```diff
@@ -15,5 +15,9 @@ export default ({ route, setRoute }, { navId, isInline }) => ({
   isVisible: isInline ? isPast(navId, route) : isVisible(navId, route),
   isInline,
   isStack: !isInline,
-  isPast: isPast(navId, route)
+  isPast: isPast(navId, route),
+  isActive: isPast(itemId || navId, activeRoute.slice(1, activeRoute.length - 1)),
+  setActiveRoute: () => {
+    setActiveRoute(absoluteRoute);
+  }
```
[the remark this answers is not in the document]
Comment (edwinjoseph) [reply src/utils/build-render-props.js]: 👍

File: example/src/App.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisvxd) [reply example/src/App.js]: Too much for now, let's just leave as is.

**Pull request merged**